rootex/utility/component_array_iterator.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include "common/types.h"
+
+template <typename DataType>
+class ComponentArrayIterator
+{
+protected:
+	typename Vector<DataType>::iterator m_Itr;
+	int m_Index;
+	Vector<bool>* m_IsValid;
+
+public:
+	using iterator_category = std::random_access_iterator_tag;
+	using value_type = DataType;
+	using difference_type = std::ptrdiff_t;
+	using pointer = DataType*;
+	using reference = DataType&;
+
+	ComponentArrayIterator(Vector<bool>& isValid, typename Vector<DataType>::iterator itr)
+	{
+		m_IsValid = &isValid;
+		m_Itr = itr;
+		m_Index = 0;
+	}
+	ComponentArrayIterator(typename Vector<DataType>::iterator itr)
+	{
+		m_Itr = itr;
+		m_Index = 0;
+	}
+	ComponentArrayIterator(const ComponentArrayIterator<DataType>& rawIterator) = default; // Copy constructor
+	~ComponentArrayIterator() { }
+
+	ComponentArrayIterator<DataType>& operator=(const ComponentArrayIterator<DataType>& rawIterator) = default;
+	ComponentArrayIterator<DataType>& operator=(typename Vector<DataType>::iterator itr)
+	{
+		m_Itr = itr;
+		return *this;
+	}
```

Comment:
Do we really need this?
Using `Vector` arguments in `ComponentArrayIterator` methods doesn't seem like something we should be allowing. The entire point of `ComponentArray` was to abstract away all the vector details